diff --git i/src/config.h w/src/config.h
index 4fd39cc..8b9f056 100644
--- i/src/config.h
+++ w/src/config.h
@@ -98,3 +98,5 @@
 
 /* Define to the directory where all game files are located */
 #define FILES_DIR "."
+
+#define SYS_BIG_ENDIAN
diff --git i/src/doomgeneric_ascii.c w/src/doomgeneric_ascii.c
index fcefb0b..ee15ce5 100644
--- i/src/doomgeneric_ascii.c
+++ w/src/doomgeneric_ascii.c
@@ -364,6 +364,10 @@ static inline unsigned char convertCsiToDoomKey(const char **const buf)
 		return KEY_DOWNARROW;
 	case 'C':
 		return KEY_RIGHTARROW;
+	case ' ':
+		return KEY_FIRE;
+	case 'E':
+		return KEY_USE;
 	case 'D':
 		return KEY_LEFTARROW;
 	case 'H':
@@ -439,6 +443,8 @@ static inline unsigned char convertToDoomKey(const char **const buf)
 	switch (**buf) {
 	case '\012':
 		return KEY_ENTER;
+	case ' ':
+		return KEY_FIRE;
 	case '\033':
 		switch (*((*buf) + 1)) {
 		case '[':
diff --git i/src/i_swap.h w/src/i_swap.h
index d1fd754..841e0c3 100644
--- i/src/i_swap.h
+++ w/src/i_swap.h
@@ -20,8 +20,8 @@
 #ifndef __I_SWAP__
 #define __I_SWAP__
 
-#ifdef ORIGCODE
-#include "SDL_endian.h"
+#if 1 //ifdef ORIGCODE
+//#include "SDL_endian.h"
 
 // Endianess handling.
 // WAD files are stored little endian.
@@ -31,8 +31,8 @@
 // These are deliberately cast to signed values; this is the behaviour
 // of the macros in the original source and some code relies on it.
 
-#define SHORT(x)  ((signed short) SDL_SwapLE16(x))
-#define LONG(x)   ((signed int) SDL_SwapLE32(x))
+#define SHORT(x)  ((signed short) __builtin_bswap16(x))
+#define LONG(x)   ((signed int) __builtin_bswap32(x))
 
 // Defines for checking the endianness of the system.
 
